In [86]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

In [87]:
# csv files path

imdb_data_path = 'C:/Users/KAWAN/Desktop/aulas_python_pimes/imdb_data.xlsx'
runtime_path = 'C:/Users/KAWAN/Desktop/aulas_python_pimes/runtime.xlsx'

# pickle files path

imdb_data_pkl_path = 'C:/Users/KAWAN/Desktop/aulas_python_pimes/imdb_data.pkl'
runtime__pkl_path = 'C:/Users/KAWAN/Desktop/aulas_python_pimes/runtime.pkl'

In [88]:
imdb_data = pd.read_excel(imdb_data_path)
runtime_data = pd.read_excel(runtime_path)

#### IMDB movies data

In [90]:
display(imdb_data)

Name    Year  User Rating  Critic Rating  \
0                      Superman  2025.0          7.4           68.0   
1       Jurassic World: Rebirth  2025.0          6.0           50.0   
2             War of the Worlds  2025.0          2.6            6.0   
3             The Life of Chuck  2024.0          7.4           67.0   
4                    Descendent  2025.0          4.9            NaN   
...                         ...     ...          ...            ...   
3794              The Dead Link  2002.0          7.2            NaN   
3795  Uchuujin from Outer Space  2007.0          5.5            NaN   
3796                The Complex  2012.0          5.5            NaN   
3797    Epitaph: Bread and Salt  2013.0          4.4            NaN   
3798         Trash and Progress  2012.0          4.6            NaN   

                                                 Genres     Duration  \
0     Adventure Epic, Globetrotting Adventure, Sci-F...  Cast & crew   
1     Dinosaur Adventure, Action, Adventure, Sci-Fi,...  Cast & crew   
2              Alien Invasion, Horror, Sci-Fi, Thriller  Cast & crew   
3                                Drama, Fantasy, Sci-Fi  Cast & crew   
4              Drama, Horror, Mystery, Sci-Fi, Thriller  Cast & crew   
...                                                 ...          ...   
3794                                             Sci-Fi  Cast & crew   
3795                             Sitcom, Comedy, Sci-Fi  Cast & crew   
3796                             Comedy, Horror, Sci-Fi  Cast & crew   
3797                             Action, Horror, Sci-Fi  Cast & crew   
3798                             Drama, Sci-Fi, Western  Cast & crew   

     Age Rating  Popularity          Director  \
0         PG-13    236000.0        James Gunn   
1         PG-13    112000.0    Gareth Edwards   
2         PG-13     23000.0          Rich Lee   
3             R     22000.0     Mike Flanagan   
4           NaN       588.0     Peter Cilella   
...         ...         ...               ...   
3794        NaN        58.0          Ben Juhl   
3795        NaN        85.0    Danny Mann Jr.   
3796        NaN        57.0      Josh Spiegel   
3797        NaN        81.0    Nathyn Masters   
3798  Not Rated        55.0  Abraham Dieckman   

                                   Production Companies  
0     ['DC Studios', 'Domain Entertainment (II)', 'T...  
1     ['Universal Pictures', 'Amblin Entertainment',...  
2     ['Universal Pictures', 'Bazelevs Entertainment...  
3     ['Neon', 'Intrepid Pictures', 'Red Room Pictur...  
4                                      ['Rustic Films']  
...                                                 ...  
3794                     ['Twilightshow Entertainment']  
3795  ["Don't Give A Damn Productions", 'Troma Enter...  
3796                                                NaN  
3797                             ['TimeCode Mechanics']  
3798                                                NaN  

[3799 rows x 10 columns]

In [91]:
# repeated lines in IMDB data?

duplic = imdb_data.value_counts()
duplic = duplic[duplic > 1]  
duplic_titles = duplic.index.tolist()

print(len(duplic_titles))

0


In [92]:
# repeated titles?

duplic_ = imdb_data['Name'].value_counts()
duplic_ = duplic_[duplic_ > 1]  
duplic_titles = duplic_.index.tolist()

print(duplic_titles)
print(len(duplic_titles))

['Dr. Jekyll and Mr. Hyde', 'Superman', 'Frankenstein', 'Journey to the Center of the Earth', 'Mars', 'The Avengers', 'The Whisperer in Darkness', 'The Arrival', 'Meteor', 'End of the World', 'Captain America', 'Blake of Scotland Yard', 'The Hive', 'Firestarter', "It's Alive", 'Black Friday', 'Project X', 'Alien Predator', 'The Sighting', 'The Lost World', 'Chrysalis', 'Flatliners', 'Prism', 'The Thing', 'Savage', 'The Survivalist', 'Hellboy', 'Ghostbusters', 'The Time Machine', 'Buck Rogers', 'Paradox', 'Parallel', 'Air', 'RoboCop', 'Alien Abduction', 'Oblivion', '20,000 Leagues Under the Sea', 'Invasion of the Body Snatchers', 'Pandemic', 'Mortal Kombat', 'Interface', 'Not of This Earth', 'The Silencers', 'The Stepford Wives', 'Mysterious Island', 'The Black Hole', 'Invaders from Mars', 'The Day the Earth Stood Still', 'Unidentified', 'The Invisible Man', 'Planet of the Apes', 'Death House', 'Lilo & Stitch', 'The Nutty Professor', 'Spiders', "The Brain That Wouldn't Die", 'The Fly', 

There are 75 titles repeated

In [94]:
# why exist repeated titles? example with 'Superman' movie. I checked for others repeated titles.

superman = imdb_data[imdb_data['Name']=='Superman']
display(superman)

# different 'Superman' movies - the 2025, 1973 and 1948 movies - has the same title in the dataframe

Name    Year  User Rating  Critic Rating  \
0     Superman  2025.0          7.4           68.0   
1269  Superman  1948.0          6.7            NaN   
2486  Superman  1973.0          6.4            NaN   

                                                 Genres     Duration  \
0     Adventure Epic, Globetrotting Adventure, Sci-F...  Cast & crew   
1269                          Superhero, Action, Sci-Fi  Cast & crew   
2486               Superhero, Action, Adventure, Sci-Fi  Cast & crew   

     Age Rating  Popularity               Director  \
0         PG-13    236000.0             James Gunn   
1269   Approved         1.3  Spencer Gordon Bennet   
2486        NaN       224.0   Robert Leslie Bellem   

                                   Production Companies  
0     ['DC Studios', 'Domain Entertainment (II)', 'T...  
1269                        ['Sam Katzman Productions']  
2486                                   ['Warner Bros.']

In [95]:
# rename the column 'Name'

imdb_data = imdb_data.rename(columns = {'Name':'title'})
imdb_data.columns

Index(['title', 'Year', 'User Rating', 'Critic Rating', 'Genres', 'Duration',
       'Age Rating', 'Popularity', 'Director', 'Production Companies'],
      dtype='object')

In [96]:
# whats the line of 'Pandemic' movie? both films has the same duration, what may induce confusion.

pandemic_line = imdb_data[imdb_data['title'] =='Pandemic']
display(pandemic_line)

# there are two different movies

title    Year  User Rating  Critic Rating  \
1345  Pandemic  2016.0          4.4           47.0   
2118  Pandemic  2009.0          3.4            NaN   

                                                 Genres     Duration  \
1345  B-Action, B-Horror, Action, Horror, Sci-Fi, Th...  Cast & crew   
2118       Conspiracy Thriller, Drama, Sci-Fi, Thriller  Cast & crew   

     Age Rating  Popularity       Director  \
1345      TV-MA         8.7     John Suits   
2118          R       918.0  Jason Connery   

                                   Production Companies  
1345  ['New Artists Alliance', 'Tadross Media Group'...  
2118  ['Alianza Films International', 'Hollywood Pho...

In [97]:
imdb_data.dtypes

title                    object
Year                    float64
User Rating             float64
Critic Rating           float64
Genres                   object
Duration                 object
Age Rating               object
Popularity              float64
Director                 object
Production Companies     object
dtype: object

In [98]:
# year as integer

imdb_data["Year"] = imdb_data["Year"].astype("Int64")   

In [99]:
imdb_data.dtypes

title                    object
Year                      Int64
User Rating             float64
Critic Rating           float64
Genres                   object
Duration                 object
Age Rating               object
Popularity              float64
Director                 object
Production Companies     object
dtype: object

In [100]:
imdb_data.head(5)

title  Year  User Rating  Critic Rating  \
0                 Superman  2025          7.4           68.0   
1  Jurassic World: Rebirth  2025          6.0           50.0   
2        War of the Worlds  2025          2.6            6.0   
3        The Life of Chuck  2024          7.4           67.0   
4               Descendent  2025          4.9            NaN   

                                              Genres     Duration Age Rating  \
0  Adventure Epic, Globetrotting Adventure, Sci-F...  Cast & crew      PG-13   
1  Dinosaur Adventure, Action, Adventure, Sci-Fi,...  Cast & crew      PG-13   
2           Alien Invasion, Horror, Sci-Fi, Thriller  Cast & crew      PG-13   
3                             Drama, Fantasy, Sci-Fi  Cast & crew          R   
4           Drama, Horror, Mystery, Sci-Fi, Thriller  Cast & crew        NaN   

   Popularity        Director  \
0    236000.0      James Gunn   
1    112000.0  Gareth Edwards   
2     23000.0        Rich Lee   
3     22000.0   Mike Flanagan   
4       588.0   Peter Cilella   

                                Production Companies  
0  ['DC Studios', 'Domain Entertainment (II)', 'T...  
1  ['Universal Pictures', 'Amblin Entertainment',...  
2  ['Universal Pictures', 'Bazelevs Entertainment...  
3  ['Neon', 'Intrepid Pictures', 'Red Room Pictur...  
4                                   ['Rustic Films']

In [101]:
# i will create a new column with all the titles, but the repeated titles will be followed by the year. Its a way to distinguish between movies with
# the same title.

imdb_data1 = imdb_data.copy()
counts = imdb_data1["title"].value_counts(dropna=False)

def adjust_title(row):
    title = row["title"]
    year = row["Year"]
    if pd.isna(title):   # if has no title, keep as NaN
        return title
    if counts[title] > 1:
        return f"{title} {int(year)}" if pd.notna(year) else title
    else:
        return title

imdb_data1["title_adjusted"] = imdb_data1.apply(adjust_title, axis=1)

In [102]:
# reordering the columns

imdb_data1 = imdb_data1[['title','Year','title_adjusted','User Rating','Critic Rating','Genres','Duration','Age Rating','Popularity','Director','Production Companies']]
display(imdb_data1)

title  Year             title_adjusted  User Rating  \
0                      Superman  2025              Superman 2025          7.4   
1       Jurassic World: Rebirth  2025    Jurassic World: Rebirth          6.0   
2             War of the Worlds  2025     War of the Worlds 2025          2.6   
3             The Life of Chuck  2024          The Life of Chuck          7.4   
4                    Descendent  2025                 Descendent          4.9   
...                         ...   ...                        ...          ...   
3794              The Dead Link  2002              The Dead Link          7.2   
3795  Uchuujin from Outer Space  2007  Uchuujin from Outer Space          5.5   
3796                The Complex  2012                The Complex          5.5   
3797    Epitaph: Bread and Salt  2013    Epitaph: Bread and Salt          4.4   
3798         Trash and Progress  2012         Trash and Progress          4.6   

      Critic Rating                                             Genres  \
0              68.0  Adventure Epic, Globetrotting Adventure, Sci-F...   
1              50.0  Dinosaur Adventure, Action, Adventure, Sci-Fi,...   
2               6.0           Alien Invasion, Horror, Sci-Fi, Thriller   
3              67.0                             Drama, Fantasy, Sci-Fi   
4               NaN           Drama, Horror, Mystery, Sci-Fi, Thriller   
...             ...                                                ...   
3794            NaN                                             Sci-Fi   
3795            NaN                             Sitcom, Comedy, Sci-Fi   
3796            NaN                             Comedy, Horror, Sci-Fi   
3797            NaN                             Action, Horror, Sci-Fi   
3798            NaN                             Drama, Sci-Fi, Western   

         Duration Age Rating  Popularity          Director  \
0     Cast & crew      PG-13    236000.0        James Gunn   
1     Cast & crew      PG-13    112000.0    Gareth Edwards   
2     Cast & crew      PG-13     23000.0          Rich Lee   
3     Cast & crew          R     22000.0     Mike Flanagan   
4     Cast & crew        NaN       588.0     Peter Cilella   
...           ...        ...         ...               ...   
3794  Cast & crew        NaN        58.0          Ben Juhl   
3795  Cast & crew        NaN        85.0    Danny Mann Jr.   
3796  Cast & crew        NaN        57.0      Josh Spiegel   
3797  Cast & crew        NaN        81.0    Nathyn Masters   
3798  Cast & crew  Not Rated        55.0  Abraham Dieckman   

                                   Production Companies  
0     ['DC Studios', 'Domain Entertainment (II)', 'T...  
1     ['Universal Pictures', 'Amblin Entertainment',...  
2     ['Universal Pictures', 'Bazelevs Entertainment...  
3     ['Neon', 'Intrepid Pictures', 'Red Room Pictur...  
4                                      ['Rustic Films']  
...                                                 ...  
3794                     ['Twilightshow Entertainment']  
3795  ["Don't Give A Damn Productions", 'Troma Enter...  
3796                                                NaN  
3797                             ['TimeCode Mechanics']  
3798                                                NaN  

[3799 rows x 11 columns]

In [103]:
# keep having repeated titles?

duplic__ = imdb_data1['title_adjusted'].value_counts()
duplic__ = duplic__[duplic__ > 1]  
duplic_titles_ = duplic__.index.tolist()

print(duplic_titles_)
print(len(duplic_titles_))

['Blake of Scotland Yard 1937', 'Paradox 2016']
2


In [104]:
blackofscot = imdb_data1[imdb_data1['title_adjusted'] =='Blake of Scotland Yard 1937']
paradox = imdb_data1[imdb_data1['title_adjusted'] =='Paradox 2016']
display(blackofscot, paradox)

title  Year               title_adjusted  User Rating  \
3137  Blake of Scotland Yard  1937  Blake of Scotland Yard 1937          5.0   
3661  Blake of Scotland Yard  1937  Blake of Scotland Yard 1937          3.3   

      Critic Rating                    Genres     Duration Age Rating  \
3137            NaN  Adventure, Crime, Sci-Fi  Cast & crew   Approved   
3661            NaN             Crime, Sci-Fi  Cast & crew   Approved   

      Popularity        Director              Production Companies  
3137       211.0  Robert F. Hill  ['Victory Pictures Corporation']  
3661       140.0  Robert F. Hill  ['Victory Pictures Corporation']

title  Year title_adjusted  User Rating  Critic Rating  \
2086  Paradox  2016   Paradox 2016          4.6            NaN   
2923  Paradox  2016   Paradox 2016          6.1            NaN   

                                Genres     Duration Age Rating  Popularity  \
2086             Action, Drama, Sci-Fi  Cast & crew      TV-MA         3.4   
2923  Drama, Romance, Sci-Fi, Thriller  Cast & crew        NaN       164.0   

           Director                               Production Companies  
2086  Michael Hurst  ['Benattar/Thomas Productions', 'Parkside Pict...  
2923    Stevo Chang                          ['Organized Chaos Films']

There is two different **Paradox** film in 2016. I will exclude **Blake of Scotland Yard**.

In [106]:
imdb_data1 = imdb_data1.drop([3137, 3661])
imdb_data1 = imdb_data1.reset_index(drop=True)     # reset index
display(imdb_data1)

title  Year             title_adjusted  User Rating  \
0                      Superman  2025              Superman 2025          7.4   
1       Jurassic World: Rebirth  2025    Jurassic World: Rebirth          6.0   
2             War of the Worlds  2025     War of the Worlds 2025          2.6   
3             The Life of Chuck  2024          The Life of Chuck          7.4   
4                    Descendent  2025                 Descendent          4.9   
...                         ...   ...                        ...          ...   
3792              The Dead Link  2002              The Dead Link          7.2   
3793  Uchuujin from Outer Space  2007  Uchuujin from Outer Space          5.5   
3794                The Complex  2012                The Complex          5.5   
3795    Epitaph: Bread and Salt  2013    Epitaph: Bread and Salt          4.4   
3796         Trash and Progress  2012         Trash and Progress          4.6   

      Critic Rating                                             Genres  \
0              68.0  Adventure Epic, Globetrotting Adventure, Sci-F...   
1              50.0  Dinosaur Adventure, Action, Adventure, Sci-Fi,...   
2               6.0           Alien Invasion, Horror, Sci-Fi, Thriller   
3              67.0                             Drama, Fantasy, Sci-Fi   
4               NaN           Drama, Horror, Mystery, Sci-Fi, Thriller   
...             ...                                                ...   
3792            NaN                                             Sci-Fi   
3793            NaN                             Sitcom, Comedy, Sci-Fi   
3794            NaN                             Comedy, Horror, Sci-Fi   
3795            NaN                             Action, Horror, Sci-Fi   
3796            NaN                             Drama, Sci-Fi, Western   

         Duration Age Rating  Popularity          Director  \
0     Cast & crew      PG-13    236000.0        James Gunn   
1     Cast & crew      PG-13    112000.0    Gareth Edwards   
2     Cast & crew      PG-13     23000.0          Rich Lee   
3     Cast & crew          R     22000.0     Mike Flanagan   
4     Cast & crew        NaN       588.0     Peter Cilella   
...           ...        ...         ...               ...   
3792  Cast & crew        NaN        58.0          Ben Juhl   
3793  Cast & crew        NaN        85.0    Danny Mann Jr.   
3794  Cast & crew        NaN        57.0      Josh Spiegel   
3795  Cast & crew        NaN        81.0    Nathyn Masters   
3796  Cast & crew  Not Rated        55.0  Abraham Dieckman   

                                   Production Companies  
0     ['DC Studios', 'Domain Entertainment (II)', 'T...  
1     ['Universal Pictures', 'Amblin Entertainment',...  
2     ['Universal Pictures', 'Bazelevs Entertainment...  
3     ['Neon', 'Intrepid Pictures', 'Red Room Pictur...  
4                                      ['Rustic Films']  
...                                                 ...  
3792                     ['Twilightshow Entertainment']  
3793  ["Don't Give A Damn Productions", 'Troma Enter...  
3794                                                NaN  
3795                             ['TimeCode Mechanics']  
3796                                                NaN  

[3797 rows x 11 columns]

#### Top 10 mean popularity by Director

In [108]:
# top 10 populars (by the sum of popularity)
top10 = (                                                 
    imdb_data1.groupby("Director")["Popularity"]
    .sum()
    .nlargest(10)
    .index)

stats = ['count', 'mean', 'min', 'max']
top10_directors = (
    imdb_data1[imdb_data1["Director"].isin(top10)]
    .groupby("Director")["Popularity"]
    .agg(stats).round()
    .sort_values("mean", ascending=False)
)


display(top10_directors)

count      mean       min        max
Director                                              
Denis Villeneuve      4  796750.0  659000.0   973000.0
Jon Watts             3  783667.0  610000.0   972000.0
J.J. Abrams           5  608000.0  376000.0  1000000.0
Sam Raimi             5  593800.0   73000.0   936000.0
Francis Lawrence      5  542200.0  168000.0   858000.0
George Lucas          5  531200.0       2.0   903000.0
Zack Snyder           8  448000.0   61000.0   850000.0
Michael Bay           7  417714.0  180000.0   703000.0
Steven Spielberg      9  346055.0       1.0   608000.0
Roland Emmerich       7  309143.0  100000.0   634000.0

#### Runtime Data

In [110]:
display(runtime_data)

title  duration
0                      Superman     129.0
1       Jurassic World: Rebirth     133.0
2             War of the Worlds      91.0
3             The Life of Chuck     111.0
4                    Descendent      96.0
...                         ...       ...
3797              The Dead Link      85.0
3798  Uchuujin from Outer Space     165.0
3799                The Complex      93.0
3800    Epitaph: Bread and Salt      90.0
3801         Trash and Progress      94.0

[3802 rows x 2 columns]

In [111]:
# repeated lines?

dup_runtime = runtime_data.value_counts()
dup_runtime = dup_runtime[dup_runtime > 1]  
dup_runtime_titles = dup_runtime.index.tolist()

print(dup_runtime_titles)
print(len(dup_runtime_titles))

[('Pandemic', 91.0)]
1


In [112]:
# duplicated titles?

duplic_runtime = runtime_data['title'].value_counts()
duplic_runtime = duplic_runtime[duplic_runtime > 1]  
duplic_runtime_titles = duplic_runtime.index.tolist()

print(duplic_runtime_titles)
print(len(duplic_runtime_titles))

['Dr. Jekyll and Mr. Hyde', 'Superman', 'Frankenstein', 'Journey to the Center of the Earth', 'Prism', 'The Avengers', 'The Whisperer in Darkness', 'The Arrival', 'Meteor', 'End of the World', 'Captain America', 'Blake of Scotland Yard', 'The Hive', 'Firestarter', 'Black Friday', "It's Alive", 'Project X', 'Alien Predator', 'Chrysalis', 'The Lost World', 'The Sighting', 'Mars', 'Flatliners', 'The Thing', 'Savage', 'The Survivalist', 'The Day the Earth Stood Still', 'Ghostbusters', 'The Time Machine', 'Paradox', 'Buck Rogers', 'Parallel', 'Air', 'RoboCop', 'Alien Abduction', 'Oblivion', '20,000 Leagues Under the Sea', 'Pandemic', 'Invasion of the Body Snatchers', 'Mortal Kombat', 'Interface', 'The Silencers', 'Not of This Earth', 'Mysterious Island', 'The Stepford Wives', 'The Black Hole', 'Invaders from Mars', 'Hellboy', 'Unidentified', 'The Invisible Man', 'Muse', 'Planet of the Apes', 'Spiders', "The Brain That Wouldn't Die", 'Death House', 'Stasis', 'The Toxic Avenger', 'The Island 

In [113]:
superman_ = runtime_data[runtime_data['title']=='Superman']
display(superman_)

title  duration
0     Superman     129.0
1271  Superman     244.0
2488  Superman     104.0

**imdb_data** has 3798 titles

**runtime_data** has 3801 titles

2 titles more

In [115]:
# titles in runtime_data that aren´t in imdb_data   
missing_title = runtime_data[~runtime_data['title'].isin(imdb_data1['title'])]
unique_miss = missing_title['title'].unique()

print(unique_miss)

['Jimmy and Stiggs' 'Afterburn' 'Hollywood Warrioress: The Movie'
 'Andromeda 3' 'Kamillions' 'Chrome Angels' 'Blake of Scotland Yard']


In [116]:
# titles in imdb_data that aren´t in runtime_data   
missing_title1 = imdb_data1[~imdb_data1['title'].isin(runtime_data['title'])]
unique_miss1 = missing_title1['title'].unique()

print(unique_miss1)

[nan]


In [117]:
# I will drop the movies in runtime that aren´t in the IMDB data because i dont have more information about them more than the title.

to_drop = ['Jimmy and Stiggs','Afterburn','Hollywood Warrioress: The Movie', 'Andromeda 3','Kamillions','Chrome Angels','Blake of Scotland Yard']

runtime1 = runtime_data.copy()
rows_to_drop = runtime1[runtime1['title'].isin(to_drop)].index
runtime1 = runtime1.drop(rows_to_drop).reset_index(drop=True)
display(runtime1)

title  duration
0                      Superman     129.0
1       Jurassic World: Rebirth     133.0
2             War of the Worlds      91.0
3             The Life of Chuck     111.0
4                    Descendent      96.0
...                         ...       ...
3789              The Dead Link      85.0
3790  Uchuujin from Outer Space     165.0
3791                The Complex      93.0
3792    Epitaph: Bread and Salt      90.0
3793         Trash and Progress      94.0

[3794 rows x 2 columns]

In [118]:
# the titles in imdb_data1 the arent in runtime1

comp = imdb_data1[~imdb_data1['title'].isin(runtime1['title'])]
comp_miss1 = comp['title'].unique()

print(comp_miss1)

[nan]


In [119]:
# rename the column title to merge

runtime1 = runtime1.rename(columns={'title':'title_adjusted'})
runtime1.columns

Index(['title_adjusted', 'duration'], dtype='object')

#### Missing values

In [121]:
imdb_data1[imdb_data1.isna().all(axis=1)]

title  Year title_adjusted  User Rating  Critic Rating Genres Duration  \
2987   NaN  <NA>            NaN          NaN            NaN    NaN      NaN   
2988   NaN  <NA>            NaN          NaN            NaN    NaN      NaN   
2994   NaN  <NA>            NaN          NaN            NaN    NaN      NaN   

     Age Rating  Popularity Director Production Companies  
2987        NaN         NaN      NaN                  NaN  
2988        NaN         NaN      NaN                  NaN  
2994        NaN         NaN      NaN                  NaN

In [122]:
runtime1[runtime1.isna().all(axis=1)].sum()

title_adjusted      0
duration          0.0
dtype: object

In [123]:
imdb_data1 = imdb_data1.dropna(how='all')
len(imdb_data1)

3794

In [124]:
len(runtime1)

3794

Now, both dataframe has the same dimension

In [126]:
# ordering by alphabetical order

imdb_data1 = imdb_data1.sort_values('title').reset_index(drop=True)
runtime1 = runtime1.sort_values('title_adjusted').reset_index(drop=True)

In [127]:
imdb_data1.head(10)

title  Year  \
0                      *batteries not included  1987   
1                                           +1  2013   
2                              .com for Murder  2002   
3  1,001 Ways to Enjoy the Missionary Position  2010   
4                          10 Cloverfield Lane  2016   
5                                  10,000 Days  2014   
6                              10.0 Earthquake  2014   
7                               11 Minutes Ago  2007   
8                                        11:59  2005   
9                                   12 Monkeys  1995   

                                title_adjusted  User Rating  Critic Rating  \
0                      *batteries not included          6.6           54.0   
1                                           +1          5.5           60.0   
2                              .com for Murder          2.7            NaN   
3  1,001 Ways to Enjoy the Missionary Position          4.4            NaN   
4                          10 Cloverfield Lane          7.2           76.0   
5                                  10,000 Days          2.7            NaN   
6                              10.0 Earthquake          3.5            NaN   
7                               11 Minutes Ago          6.3            NaN   
8                                        11:59          5.8            NaN   
9                                   12 Monkeys          8.0           75.0   

                                              Genres     Duration Age Rating  \
0                    Comedy, Family, Fantasy, Sci-Fi  Cast & crew         PG   
1                           Horror, Sci-Fi, Thriller  Cast & crew  Not Rated   
2                           Horror, Sci-Fi, Thriller  Cast & crew    Unrated   
3                              Comedy, Drama, Sci-Fi  Cast & crew        NaN   
4  Psychological Horror, Psychological Thriller, ...  Cast & crew      PG-13   
5                                             Sci-Fi  Cast & crew      TV-PG   
6  Disaster, Action, Adventure, Drama, Sci-Fi, Th...  Cast & crew        NaN   
7                                    Romance, Sci-Fi  Cast & crew  Not Rated   
8                          Mystery, Sci-Fi, Thriller  Cast & crew        NaN   
9  Cyberpunk, Dystopian Sci-Fi, Psychological Thr...  Cast & crew          R   

   Popularity          Director  \
0     39000.0   Matthew Robbins   
1         8.7    Dennis Iliadis   
2         5.1   Nico Mastorakis   
3       128.0          Alan Chu   
4    375000.0  Dan Trachtenberg   
5       885.0        Eric Small   
6         2.9      David Gidali   
7       445.0        Bob Gebert   
8       483.0      Jamin Winans   
9    668000.0     Terry Gilliam   

                                Production Companies  
0     ['Universal Pictures', 'Amblin Entertainment']  
1                 ['Process Films', 'Process Media']  
2                            ['Omega Entertainment']  
3                         ['1,001 Ways Productions']  
4  ['Paramount Pictures', 'Bad Robot', 'Spectrum ...  
5                          ['Bahr Productions Inc.']  
6                             ['Indy Entertainment']  
7                                 ['Fly High Films']  
8                              ['Double Edge Films']  
9  ['Universal Pictures', 'Atlas Entertainment', ...

In [128]:
runtime1.head(10)

title_adjusted  duration
0                      *batteries not included     107.0
1                                           +1      96.0
2                              .com for Murder      96.0
3  1,001 Ways to Enjoy the Missionary Position      83.0
4                          10 Cloverfield Lane     103.0
5                                  10,000 Days      91.0
6                              10.0 Earthquake      88.0
7                               11 Minutes Ago      83.0
8                                        11:59     102.0
9                                   12 Monkeys     129.0

In [129]:
# Compare value by value  to see if the titles in imdb_data are equal the titles in runtime
compare = imdb_data1['title'] == runtime1['title_adjusted']

print(f"equal values (total sum): {compare.sum()} de {len(compare)}")
print(f"different values (total sum): {compare.isna().sum() if compare.isna().any() else (~compare).sum()}")

# if differente, where?
diff = compare[~compare]
if not diff.empty:
    print("\nindex with difference values:")
    for idx in diff.index:
        print(f"index {idx}: '{imdb_data1.loc[idx, 'title']}' != '{runtime1.loc[idx, 'title_adjusted']}'")

equal values (total sum): 3794 de 3794
different values (total sum): 0


### Merge

In [189]:
imdb_finaldt = pd.merge(imdb_data1, runtime1, left_index=True, right_index=True, how='inner')
display(imdb_finaldt)

title  Year  \
0                         *batteries not included  1987   
1                                              +1  2013   
2                                 .com for Murder  2002   
3     1,001 Ways to Enjoy the Missionary Position  2010   
4                             10 Cloverfield Lane  2016   
...                                           ...   ...   
3789                                      Zéro un  2003   
3790                                  iGirlfriend  2016   
3791                         www.RachelOrmont.com  2024   
3792                      xXx: State of the Union  2005   
3793                                     Æon Flux  2005   

                                 title_adjusted_x  User Rating  Critic Rating  \
0                         *batteries not included          6.6           54.0   
1                                              +1          5.5           60.0   
2                                 .com for Murder          2.7            NaN   
3     1,001 Ways to Enjoy the Missionary Position          4.4            NaN   
4                             10 Cloverfield Lane          7.2           76.0   
...                                           ...          ...            ...   
3789                                      Zéro un          5.5            NaN   
3790                                  iGirlfriend          2.9            NaN   
3791                         www.RachelOrmont.com          5.4            NaN   
3792                      xXx: State of the Union          4.5           37.0   
3793                                     Æon Flux          5.4           36.0   

                                                 Genres     Duration  \
0                       Comedy, Family, Fantasy, Sci-Fi  Cast & crew   
1                              Horror, Sci-Fi, Thriller  Cast & crew   
2                              Horror, Sci-Fi, Thriller  Cast & crew   
3                                 Comedy, Drama, Sci-Fi  Cast & crew   
4     Psychological Horror, Psychological Thriller, ...  Cast & crew   
...                                                 ...          ...   
3789              Animation, Comedy, Drama, Sci-Fi, War  Cast & crew   
3790              Teen Comedy, Comedy, Sci-Fi, Thriller  Cast & crew   
3791                                      Drama, Sci-Fi  Cast & crew   
3792  Conspiracy Thriller, Political Thriller, Actio...  Cast & crew   
3793  Cyberpunk, One-Person Army Action, Superhero, ...  Cast & crew   

     Age Rating  Popularity          Director  \
0            PG     39000.0   Matthew Robbins   
1     Not Rated         8.7    Dennis Iliadis   
2       Unrated         5.1   Nico Mastorakis   
3           NaN       128.0          Alan Chu   
4         PG-13    375000.0  Dan Trachtenberg   
...         ...         ...               ...   
3789        NaN        62.0      Jeanne Biras   
3790  Not Rated       361.0      Scott Coones   
3791        NaN        66.0        Peter Vack   
3792      PG-13     76000.0      Lee Tamahori   
3793      PG-13    136000.0      Karyn Kusama   

                                   Production Companies  \
0        ['Universal Pictures', 'Amblin Entertainment']   
1                    ['Process Films', 'Process Media']   
2                               ['Omega Entertainment']   
3                            ['1,001 Ways Productions']   
4     ['Paramount Pictures', 'Bad Robot', 'Spectrum ...   
...                                                 ...   
3789                                     ['EuropaCorp']   
3790                    ['Shoestring Film Cooperative']   
3791  ['Fast Rainbow Films', 'Gummy Films', 'Simone ...   
3792            ['Revolution Studios', 'Original Film']   
3793  ['Paramount Pictures', 'Lakeshore Entertainmen...   

                                 title_adjusted_y  duration  
0                         *batteries not included     107.0  
1                                              +1      96.0  
2            

In [193]:
imdb_finaldt = imdb_finaldt.drop('title_adjusted_y', axis=1)            # drop this column 
imdb_finaldt = imdb_finaldt.rename(columns = {'title_adjusted_x':'titles'})
imdb_finaldt.head(2)

title  Year                   titles  User Rating  \
0  *batteries not included  1987  *batteries not included          6.6   
1                       +1  2013                       +1          5.5   

   Critic Rating                           Genres     Duration Age Rating  \
0           54.0  Comedy, Family, Fantasy, Sci-Fi  Cast & crew         PG   
1           60.0         Horror, Sci-Fi, Thriller  Cast & crew  Not Rated   

   Popularity         Director  \
0     39000.0  Matthew Robbins   
1         8.7   Dennis Iliadis   

                             Production Companies  duration  
0  ['Universal Pictures', 'Amblin Entertainment']     107.0  
1              ['Process Films', 'Process Media']      96.0

### Verification

Some verification to see if its merge is correct considering the duplicate titles.

In [191]:
superman_check = imdb_finaldt[imdb_finaldt['title']=='Superman']
superman_check

title  Year title_adjusted_x  User Rating  Critic Rating  \
2724  Superman  2025    Superman 2025          7.4           68.0   
2725  Superman  1948    Superman 1948          6.7            NaN   
2726  Superman  1973    Superman 1973          6.4            NaN   

                                                 Genres     Duration  \
2724  Adventure Epic, Globetrotting Adventure, Sci-F...  Cast & crew   
2725                          Superhero, Action, Sci-Fi  Cast & crew   
2726               Superhero, Action, Adventure, Sci-Fi  Cast & crew   

     Age Rating  Popularity               Director  \
2724      PG-13    236000.0             James Gunn   
2725   Approved         1.3  Spencer Gordon Bennet   
2726        NaN       224.0   Robert Leslie Bellem   

                                   Production Companies title_adjusted_y  \
2724  ['DC Studios', 'Domain Entertainment (II)', 'T...         Superman   
2725                        ['Sam Katzman Productions']         Superman   
2726                                   ['Warner Bros.']         Superman   

      duration  
2724     129.0  
2725     244.0  
2726     104.0

It´s right!

In [197]:
jekyll_check = imdb_finaldt[imdb_finaldt['title']=='Dr. Jekyll and Mr. Hyde']
jekyll_check

title  Year                        titles  User Rating  \
940  Dr. Jekyll and Mr. Hyde  1941  Dr. Jekyll and Mr. Hyde 1941          6.8   
941  Dr. Jekyll and Mr. Hyde  2017  Dr. Jekyll and Mr. Hyde 2017          5.7   
942  Dr. Jekyll and Mr. Hyde  1931  Dr. Jekyll and Mr. Hyde 1931          7.6   
943  Dr. Jekyll and Mr. Hyde  1920  Dr. Jekyll and Mr. Hyde 1920          6.9   

     Critic Rating                                            Genres  \
940            NaN                   Drama, Horror, Sci-Fi, Thriller   
941            NaN                                    Horror, Sci-Fi   
942           88.0  Body Horror, Supernatural Horror, Horror, Sci-Fi   
943            NaN                             Drama, Horror, Sci-Fi   

        Duration Age Rating  Popularity              Director  \
940  Cast & crew   Approved     10000.0        Victor Fleming   
941  Cast & crew        NaN       537.0  B. Luciano Barsuglia   
942  Cast & crew   Approved     17000.0      Rouben Mamoulian   
943  Cast & crew    Unrated         6.5     John S. Robertson   

                                  Production Companies  duration  
940                                         ["Loew's"]     113.0  
941  ['Koa Aloha Media', 'SoCal Entertainment and M...      83.0  
942                             ['Paramount Pictures']      98.0  
943                             ['Paramount Pictures']      69.0

It´s right!

### Genres

In [291]:
# frequncy of all combined genres
genre_combinations = imdb_finaldt['Genres'].value_counts()
display(genre_combinations.head(20))  # Top 20 

Genres
Sci-Fi                                 202
Horror, Sci-Fi                         194
Horror, Sci-Fi, Thriller               123
Action, Sci-Fi                         112
Comedy, Sci-Fi                          92
Sci-Fi, Thriller                        86
Drama, Sci-Fi                           76
Action, Sci-Fi, Thriller                64
Drama, Sci-Fi, Thriller                 60
Comedy, Horror, Sci-Fi                  53
Action, Horror, Sci-Fi                  46
Action, Adventure, Sci-Fi               44
Adventure, Sci-Fi                       25
Drama, Horror, Sci-Fi, Thriller         23
Action, Adventure, Sci-Fi, Thriller     22
Drama, Horror, Sci-Fi                   22
Horror, Mystery, Sci-Fi, Thriller       21
Drama, Mystery, Sci-Fi, Thriller        21
Horror, Mystery, Sci-Fi                 19
Drama, Mystery, Sci-Fi                  19
Name: count, dtype: int64

In [207]:
# extract all unique genres
all_gen = set()
for gen in imdb_finaldt['Genres'].dropna():
    for gen in gen.split(','):
        all_gen.add(gen.strip())
print(sorted(all_gen))
print(len(all_gen))

['Action', 'Action Epic', 'Adult Animation', 'Adventure', 'Adventure Epic', 'Alien Invasion', 'Animal Adventure', 'Animation', 'Anime', 'Artificial Intelligence', 'B-Action', 'B-Horror', 'Baseball', 'Basketball', 'Biography', 'Body Horror', 'Body Swap Comedy', 'Boxing', 'Buddy Comedy', 'Buddy Cop', 'Bumbling Detective', 'Caper', 'Car Action', 'Comedy', 'Coming-of-Age', 'Computer Animation', 'Conspiracy Thriller', 'Contemporary Western', 'Cop Drama', 'Cozy Mystery', 'Crime', 'Cyber Thriller', 'Cyberpunk', 'Dark Comedy', 'Dark Fantasy', 'Dark Romance', 'Desert Adventure', 'Dinosaur Adventure', 'Disaster', 'Docudrama', 'Drama', 'Drug Crime', 'Dystopian Sci-Fi', 'Epic', 'Erotic Thriller', 'Extreme Sport', 'Fairy Tale', 'Family', 'Fantasy', 'Fantasy Epic', 'Farce', 'Feel-Good Romance', 'Film Noir', 'Folk Horror', 'Found Footage Horror', 'Gangster', 'Globetrotting Adventure', 'Gun Fu', 'Hand-Drawn Animation', 'Heist', 'High-Concept Comedy', 'History', 'Holiday', 'Holiday Animation', 'Holiday

In [211]:
# frequency of each genres

from collections import Counter

count_gen = Counter()
for gen in imdb_finaldt['Genres'].dropna():
    gen_list = [g.strip() for g in gen.split(',')]
    count_gen.update(gen_list)

for gen, count in count_gen.most_common():
    print(f"{gen}: {count} films")

Sci-Fi: 3711 films
Horror: 1340 films
Action: 1318 films
Thriller: 1259 films
Comedy: 943 films
Adventure: 927 films
Drama: 870 films
Fantasy: 511 films
Mystery: 415 films
Family: 269 films
Superhero: 252 films
Romance: 241 films
Space Sci-Fi: 193 films
Crime: 176 films
Animation: 176 films
Dystopian Sci-Fi: 165 films
Alien Invasion: 157 films
Dark Comedy: 144 films
Body Horror: 141 films
Parody: 136 films
B-Horror: 135 films
Time Travel: 98 films
Monster Horror: 90 films
Supernatural Horror: 75 films
Slapstick: 75 films
Sci-Fi Epic: 73 films
Satire: 73 films
Artificial Intelligence: 70 films
Computer Animation: 69 films
Kaiju: 56 films
Urban Adventure: 54 films
Cyberpunk: 53 films
Psychological Thriller: 50 films
Action Epic: 47 films
Zombie Horror: 46 films
Dark Fantasy: 43 films
Quest: 43 films
Psychological Drama: 43 films
B-Action: 42 films
Musical: 42 films
Hand-Drawn Animation: 41 films
Dinosaur Adventure: 40 films
Adventure Epic: 39 films
Supernatural Fantasy: 38 films
Suspense

In [233]:
# a dummy for each genre

genres_dummies = imdb_finaldt['Genres'].str.get_dummies(sep=', ')

imdb_finaldt = pd.concat([imdb_finaldt, genres_dummies], axis=1)
print(imdb_finaldt.head())

                                         title  Year  \
0                      *batteries not included  1987   
1                                           +1  2013   
2                              .com for Murder  2002   
3  1,001 Ways to Enjoy the Missionary Position  2010   
4                          10 Cloverfield Lane  2016   

                                        titles  User Rating  Critic Rating  \
0                      *batteries not included          6.6           54.0   
1                                           +1          5.5           60.0   
2                              .com for Murder          2.7            NaN   
3  1,001 Ways to Enjoy the Missionary Position          4.4            NaN   
4                          10 Cloverfield Lane          7.2           76.0   

                                              Genres     Duration Age Rating  \
0                    Comedy, Family, Fantasy, Sci-Fi  Cast & crew         PG   
1                           Horror

In [235]:
for cols in imdb_finaldt:
    print(cols)

title
Year
titles
User Rating
Critic Rating
Genres
Duration
Age Rating
Popularity
Director
Production Companies
duration
Action
Action Epic
Adult Animation
Adventure
Adventure Epic
Alien Invasion
Animal Adventure
Animation
Anime
Artificial Intelligence
B-Action
B-Horror
Baseball
Basketball
Biography
Body Horror
Body Swap Comedy
Boxing
Buddy Comedy
Buddy Cop
Bumbling Detective
Caper
Car Action
Comedy
Coming-of-Age
Computer Animation
Conspiracy Thriller
Contemporary Western
Cop Drama
Cozy Mystery
Crime
Cyber Thriller
Cyberpunk
Dark Comedy
Dark Fantasy
Dark Romance
Desert Adventure
Dinosaur Adventure
Disaster
Docudrama
Drama
Drug Crime
Dystopian Sci-Fi
Epic
Erotic Thriller
Extreme Sport
Fairy Tale
Family
Fantasy
Fantasy Epic
Farce
Feel-Good Romance
Film Noir
Folk Horror
Found Footage Horror
Gangster
Globetrotting Adventure
Gun Fu
Hand-Drawn Animation
Heist
High-Concept Comedy
History
Holiday
Holiday Animation
Holiday Comedy
Horror
Jungle Adventure
Kaiju
Kung Fu
Legal Drama
Martial Arts
Me

### Production Companies

In [289]:
# frequncy of all combined production companies
prod_comp_combinations = imdb_finaldt['Production Companies'].value_counts()
display(prod_comp_combinations.head(40))       # top 30

Production Companies
[]                                                                 232
[Universal Pictures]                                                39
[Paramount Pictures]                                                28
[Republic Pictures]                                                 23
[Sam Katzman Productions]                                           19
[Walt Disney Productions]                                           16
[Polonia Brothers Entertainment]                                    15
[Marvel Studios]                                                    15
[Universal International Pictures (UI)]                             14
[The Asylum, Acme Holding Company]                                  12
[Metro-Goldwyn-Mayer (MGM)]                                         11
[Full Moon Entertainment]                                           10
[Warner Bros.]                                                       9
[Twentieth Century Fox]                                 

In [225]:
import ast

# Converter string to list
imdb_finaldt['Production Companies'] = imdb_finaldt['Production Companies'].apply(
    lambda x: ast.literal_eval(x) if isinstance(x, str) else x
)

all_producers = imdb_finaldt['Production Companies'].explode()
unique_producers = all_producers.dropna().unique()                         # each production company individually
print(unique_producers)

producers_count = all_producers.value_counts()                             # count
print('In how much films each production company appears (just the 20 most):')
print(producers_count.head(20))      

['Universal Pictures' 'Amblin Entertainment' 'Process Films' ...
 'Shoestring Film Cooperative' 'Fast Rainbow Films' 'Simone Films']
In how much films each production company appears (just the 20 most):
Production Companies
Paramount Pictures           101
Universal Pictures            99
Warner Bros.                  90
Twentieth Century Fox         75
The Asylum                    69
Columbia Pictures             68
Walt Disney Pictures          39
New Line Cinema               36
Acme Holding Company          34
Metro-Goldwyn-Mayer (MGM)     34
Marvel Studios                34
Republic Pictures             31
DC Entertainment              28
Amblin Entertainment          25
Marvel Entertainment          24
Warner Bros. Animation        23
Sam Katzman Productions       20
Full Moon Entertainment       19
Lionsgate                     19
DreamWorks Pictures           18
Name: count, dtype: int64


In [245]:
print('Total numbers of Production Companies:')
print(len(unique_producers))

Total numbers of Production Companies:
4045


In [239]:
import ast
from sklearn.preprocessing import MultiLabelBinarizer

# Converter strings para listas, se necessário
imdb_finaldt['Production Companies'] = imdb_finaldt['Production Companies'].apply(
    lambda x: ast.literal_eval(x) if isinstance(x, str) else x
)

# Substituir NaN por lista vazia
imdb_finaldt['Production Companies'] = imdb_finaldt['Production Companies'].apply(
    lambda x: x if isinstance(x, list) else []
)

# Criar as colunas dummies
mlb = MultiLabelBinarizer()
prod_dummies = pd.DataFrame(
    mlb.fit_transform(imdb_finaldt['Production Companies']),
    columns=mlb.classes_,
    index=imdb_finaldt.index
)

# Juntar ao df original
imdb_finaldt = pd.concat([imdb_finaldt, prod_dummies], axis=1)

print(imdb_finaldt.head())

                                         title  Year  \
0                      *batteries not included  1987   
1                                           +1  2013   
2                              .com for Murder  2002   
3  1,001 Ways to Enjoy the Missionary Position  2010   
4                          10 Cloverfield Lane  2016   

                                        titles  User Rating  Critic Rating  \
0                      *batteries not included          6.6           54.0   
1                                           +1          5.5           60.0   
2                              .com for Murder          2.7            NaN   
3  1,001 Ways to Enjoy the Missionary Position          4.4            NaN   
4                          10 Cloverfield Lane          7.2           76.0   

                                              Genres     Duration Age Rating  \
0                    Comedy, Family, Fantasy, Sci-Fi  Cast & crew         PG   
1                           Horror

In [277]:
display(imdb_finaldt[['Production Companies', 'Genres', 'Horror', 'Sci-Fi', 'Universal Pictures', 'Paramount Pictures']].head(20))

Production Companies  \
0          [Universal Pictures, Amblin Entertainment]   
1                      [Process Films, Process Media]   
2                               [Omega Entertainment]   
3                            [1,001 Ways Productions]   
4   [Paramount Pictures, Bad Robot, Spectrum Effects]   
5                             [Bahr Productions Inc.]   
6                                [Indy Entertainment]   
7                                    [Fly High Films]   
8                                 [Double Edge Films]   
9   [Universal Pictures, Atlas Entertainment, Clas...   
10         [Columbia Pictures, Luna Productions Inc.]   
11  [Super Jupiter Films, Wolfpak Industry Product...   
12                          [Morningside Productions]   
13                                     [Kooroc Films]   
14                          [Walt Disney Productions]   
15            [Williamson Submarine Film Corporation]   
16                        [Metro-Goldwyn-Mayer (MGM)]   
17  [Columbia Pictures, Centropolis Entertainment,...   
18                                                 []   
19                                      [Wesely Bros]   

                                               Genres  Horror  Sci-Fi  \
0                     Comedy, Family, Fantasy, Sci-Fi       0       1   
1                            Horror, Sci-Fi, Thriller       1       1   
2                            Horror, Sci-Fi, Thriller       1       1   
3                               Comedy, Drama, Sci-Fi       0       1   
4   Psychological Horror, Psychological Thriller, ...       1       1   
5                                              Sci-Fi       0       1   
6   Disaster, Action, Adventure, Drama, Sci-Fi, Th...       0       1   
7                                     Romance, Sci-Fi       0       1   
8                           Mystery, Sci-Fi, Thriller       0       1   
9   Cyberpunk, Dystopian Sci-Fi, Psychological Thr...       0       1   
10  Action, Adventure, Drama, Mystery, Sci-Fi, Thr...       0       1   
11                             Drama, Mystery, Sci-Fi       0       1   
12  Adventure Epic, Kaiju, Monster Horror, Sci-Fi ...       1       1   
13                             Drama, Fantasy, Sci-Fi       0       1   
14  Globetrotting Adventure, Sea Adventure, Steamp...       0       1   
15                          Action, Adventure, Sci-Fi       0       1   
16  Artificial Intelligence, Sci-Fi Epic, Space Sc...       0       1   
17      Disaster, Survival, Action, Adventure, Sci-Fi       0       1   
18                Action, Adventure, Sci-Fi, Thriller       0       1   
19                Dystopian Sci-Fi, Adventure, Sci-Fi       0       1   

    Universal Pictures  Paramount Pictures  
0                    1                   0  
1                    0                   0  
2                    0                   0  
3                    0                   0  
4                    0                   1  
5                    0                   0  
6                    0                   0  
7                    0                   0  
8                    0                   0  
9                    1                   0  
10                   0                   0  
11                   0                   0  
12                   0                   0  
13                   0                   0  
14                   0                   0  
15                   0                   0  
16                   0                   0  
17                   0                   0  
18                   0                   0  
19                   0                   0

### Complete Dataframe

In [ ]:
# save as excel file

#path1 = 'C:/Users/KAWAN/Desktop/aulas_python_pimes/imdb_finaldt.xlsx'
#imdb_finaldt.to_excel(path1, index=False)

## Descriptive Statistics

##### Top 10 best films (by User Rating)

In [368]:
top10_film_user = imdb_finaldt.nlargest(10, 'User Rating')[['title_adjusted_x', 'User Rating']]
print("Top 10 Films by User Rating:")
print(top10_film_user.to_string(index=False))

Top 10 Filmes por User Rating:
                                      title_adjusted_x  User Rating
          Stranger Things: The First Shadow - Broadway          8.9
                        Tobias and the Arcane Merchant          8.9
                                                  Alya          8.8
                                             Inception          8.8
Star War the Third Gathers: The Backstroke of the West          8.8
                                Attack, Decay, Release          8.7
                                          Interstellar          8.7
        Star Wars: Episode V - The Empire Strikes Back          8.7
                                            The Matrix          8.7
                                      The Strange Dark          8.7


##### Top 10 best films (by Critic Rating)

In [380]:
top10_film_critic = imdb_finaldt.nlargest(10, 'Critic Rating')[['title_adjusted_x', 'Critic Rating']]
print('Top 10 Films by Critic Rating:')
print(top10_film_critic.to_string(index=False))

Top 10 Films by Critic Rating:
                     title_adjusted_x  Critic Rating
                Bride of Frankenstein           95.0
                               WALL·E           95.0
           E.T. the Extra-Terrestrial           92.0
  Invasion of the Body Snatchers 1956           92.0
                    Frankenstein 1931           91.0
                                  Her           91.0
   Close Encounters of the Third Kind           90.0
            It's Such a Beautiful Day           90.0
   Star Wars: Episode IV - A New Hope           90.0
Eternal Sunshine of the Spotless Mind           89.0


##### Top 10 most populars films

In [385]:
top10_pop = imdb_finaldt.nlargest(10, 'Popularity')[['title_adjusted_x','Popularity']]
print('Top 10 popular films:')
print(top10_pop.to_string(index=False))

Top 10 popular films:
                          title_adjusted_x  Popularity
Star Wars: Episode VII - The Force Awakens   1000000.0
                               The Martian    978000.0
                            Dune: Part One    973000.0
                   Avengers: Age of Ultron    972000.0
                   Spider-Man: No Way Home    972000.0
       Captain America: The Winter Soldier    946000.0
        Captain America: The First Avenger    944000.0
                                Iron Man 3    942000.0
                                Spider-Man    936000.0
                                Iron Man 2    914000.0


##### Top 10 longest films

In [387]:
top10_longest = imdb_finaldt.nlargest(10, 'duration')[['title_adjusted_x','duration']]
print('Top 10 longest films:')
print(top10_longest.to_string(index=False))

Top 10 longest films:
                          title_adjusted_x  duration
Commando Cody: Sky Marshal of the Universe     361.0
                          Girl Blood Sport     354.0
                             The Power God     300.0
             The Secret of Treasure Island     300.0
               Flash Gordon's Trip to Mars     299.0
           The Green Hornet Strikes Again!     293.0
 Captain Video: Master of the Stratosphere     287.0
                       Shadow of Chinatown     281.0
                   Mysterious Doctor Satan     267.0
                        The Phantom Creeps     265.0


##### Top 10 most productive directors

In [398]:
top10_productive = imdb_finaldt['Director'].value_counts().nlargest(10)
print(top10_productive)

Director
Mark Polonia             22
Fred Olen Ray            16
Albert Pyun              14
Spencer Gordon Bennet    14
David DeCoteau           12
Charles Band             12
Ford Beebe               11
Roger Corman             11
Bert I. Gordon           10
Robert Rodriguez         10
Name: count, dtype: int64


##### Top 10 most popular Production Companies

In [405]:
top10_pc = (                                                 
    imdb_finaldt.groupby("Production Companies")["Popularity"].sum().nlargest(10).index)

stats = ['count', 'mean', 'min', 'max']
top10_pc_ = (
    imdb_finaldt[imdb_finaldt["Production Companies"].isin(top10_pc)].groupby("Production Companies")["Popularity"].agg(stats).round().
    sort_values("mean", ascending=False))

print(top10_pc_)

                                                    count      mean       min  \
Production Companies                                                            
['Columbia Pictures', 'Marvel Enterprises', 'La...      2  847000.0  758000.0   
['Warner Bros.', 'Legendary Entertainment', 'Vi...      2  816000.0  659000.0   
['Columbia Pictures', 'Pascal Pictures', 'Marve...      2  791000.0  610000.0   
['Warner Bros.', 'RatPac-Dune Entertainment', '...      2  752000.0  725000.0   
['Columbia Pictures', 'Marvel Entertainment', '...      2  710000.0  678000.0   
['Twentieth Century Fox', 'Marvel Entertainment...      2  655500.0  559000.0   
['Paramount Pictures', 'Marvel Entertainment', ...      3  619334.0       1.0   
['Marvel Studios']                                     15  575800.0       1.0   
['Twentieth Century Fox', 'Marvel Entertainment...      3  535000.0  221000.0   
['Warner Bros.', 'Village Roadshow Pictures', '...      4  336500.0   62000.0   

                           